pkgs/development/libraries/qt-5/5.4/default.nix

```diff
@@ -152,7 +152,7 @@ let
 
       multimedia = callPackage
         (
-          { qtSubmodule, base, declarative
+          { qtSubmodule, base, declarative, pkgconfig
```

Comment:
**TL;DR** pkgconfig is a required dependency for Qt Multimedia to work correctly. Don't mark it optional.

@gebner

> What is the general plan for Qt plugins? Are they supposed to be installed system-wide?

For Qt 5.4, they are supposed to be installed system wide. In my upcoming Qt 5.5 branch, programs will be wrapped to set QT_PLUGIN_PATH semi-automatically.

> then there is a config.qtPlugins option setting QT_PLUGIN_PATH as well

I didn't know that, I will have to remove it. The `config.foo` options are deprecated.

> then there is the qt setupHook setting it as well

That is only effective at build-time (for now).

> I didn't split the packages up because enabling gstreamer support was probably intended (the dependencies were all there), it was just pkgconfig that was missing.

That is correct, missing pkgconfig was an oversight on my part. Please do not split up this package. It will be difficult or impossible for downstream package maintainers to determine if their package depends on Qt Multimedia alone or on Qt Multimedia + plugins. Consequently, we'll just have to make everything that depends on Qt Multimedia depend on the plugins also. It won't reduce the closure size, it will just make the downstream Nix expressions more verbose.

@jagajaga 

> That can matter for smone :)
> Well, indeed that is used only in one pkg, so I think that making it optional is better.

If that amount of closure size difference matters, that person shouldn't be using Qt :) (seriously; it includes WebKit and Chromium)
Those plugins are needed for Qt Multimedia to support audio output. This isn't the only package that uses it, this is just the first package that someone noticed is broken.